### ISSUE TITLE
nnUNet inference question

### ISSUE THREAD

**Issue by justingm** [MIC-DKFZ/nnUNet#376]
Hello Fabian,

I know that the patch size for inference is automatically decided by nnUNet. Is it possible to change the patch size? I wanted to use a larger patch size to see if it can improve the inference runtime.

Best,
Justin

**Comment by FabianIsensee** [comment]
Hi,
you can edit the plans file but this is not trivial because your patch size must be compatible with the number of pooling operations along each axis.
Changing the patch size after the training is completed can (and will) destroy your segmentation accuracy.
There are better ways to improve inference speed. 
`nnUNet_predict -h`:


>   -f FOLDS [FOLDS ...], --folds FOLDS [FOLDS ...]
>                         folds to use for prediction. Default is None which
>                         means that folds will be detected automatically in the
>                         model output folder


>   --disable_tta         set this flag to disable test time data augmentation
>                         via mirroring. Speeds up inference by roughly factor 4
>                         (2D) or 8 (3D)


>   --step_size STEP_SIZE
>                         don't touch

Best,
Fabian

**Comment by justingm** [comment]
Hello Fabian,

I actually already set the --disable_tta flag and I do prefer to use all five folds because this setting yields the best accuracy. With that, inference takes a little under two minutes but it would be nice if I can make it a little bit faster (due to time constraints in our application). What does the step size do exactly? I read the log file and I see that there is information about steps (x, y and z) but I don't know what this is for.

Best,
Justin

**Comment by FabianIsensee** [comment]
default is 0.5 meaning that two adjacent predictions overlap by half the patch size in each dimension. If you increase that value (up to 1, do not go higher!) then the overlap will be smaller. You can see that in action by looking at the number of tiles that are used during prediction (this is printed to stdout).
Best,
Fabian

**Comment by FabianIsensee** [comment]
increasing this can have adverse effectson segmentation quality. it is important that you double check using an evaluation on some validation set!

**Comment by justingm** [comment]
Ah yes. I do remember reading this from the paper. I will give it a try and see how it affects the accuracy of the predictions. Just to clarify, the patch size chosen in inference is the same as the patch size in the training? Sorry I'm new to this and I don't quite get why the patch size of the inference has to be compatible to the number of pooling operations in each axis.

**Comment by FabianIsensee** [comment]
the patch size is always the same.

if you pool 5 times, the patch size must be divisible by 32 so that you can divide it by 2 five times. Otherwise you will run into problems when concatenating the encoder features in the decoder

**Comment by justingm** [comment]
Ok, got it. Thanks a lot!
